src/renderer/screens/settings.tsx

```diff
@@ -0,0 +1,157 @@
+    if(ramSize === 0) {
+      plRamMin = ramData.min;
+      plRamMax = ramData.max;
+
+      if(minimum == '' && maximum == '') return;
+      if(isNaN(minvalue) || isNaN(maxvalue)) return true;
+      if (minvalue < 1024) return true;
+      else if (minvalue > maxvalue) return true;
+      else if(maxvalue > 32768) return true;
+      adjustRam(minimum, maximum, ramSize);
+    }
+  };
+  
+  // Nickname input
+  function Nickname({ value, setValue }: StateNick) {
+    const invalidName = checkNick(value);
+  
+    return (
+      <Flex display={"column"} width="60%" alignContent={"center"}>
+        <Box width={"100%"}>
+          {invalidName}
+        </Box>
+
+        <FormLabel>Nickname:</FormLabel>
+        <Input
+          placeholder={playerData.nick}
+          variant="flushed"
+          value={value}
+          onChange={(nick) => setValue(nick.target.value)}
+          isInvalid={invalidName}
+        />
+      </Flex>
+    );
+  }
+
+  // RAM input
+  function Ram({ minval, maxval, setRammn, setRammx, selects, setSelects }: StateRam) {
+    const invalidRam = checkRam(minval, maxval, selects);
+
+    return (
+      <Flex display={"column"} width='60%'>
+        {invalidRam}
+
+        <FormLabel>Ram:</FormLabel>
+        <Flex direction={"row"}>
+          <Input
+              placeholder={plRamMin}
+              variant="flushed"
+              value={minval}
+              onChange={(e) => setRammn(e.target.value)}
+              isInvalid={invalidRam}
+          />
+          <Select value={selects} onChange={e => setSelects(parseInt(e.target.value))} variant="flushed" width={"75px"} alignContent="left" id='ramType'>
+            <option value={0}>MB</option>
+            <option value={1}>GB</option>
+          </Select>
+        </Flex>
+        <Flex direction={"row"}>
+          <Input
+            placeholder={plRamMax}
+            variant="flushed"
+            value={maxval}
+            onChange={(e) => setRammx(e.target.value)}
+            isInvalid={invalidRam}
+          />
+          <Select value={selects} onChange={e => setSelects(parseInt(e.target.value))} variant="flushed" width={"75px"} id='ramType2'>
+            <option value={0}>MB</option>
+            <option value={1}>GB</option>
+          </Select>
+        </Flex>
+      </Flex>
+    );
+  }
+  
+  export default function CreateSettings() {
+    const [name, setName] = useState('');
+    const [minRam, setRammn] = useState('');
+    const [maxRam, setRammx] = useState('');
+    const [selects, setSelect] = useState(0);
+
+    return (
+      <Container>
+        <Flex margin="auto" width="500px" alignItems={"center"}>
+            <Flex width="500px" direction="row" gap="20%">
+                <Nickname setValue={setName} value={name} />
+                <Ram minval={minRam} maxval={maxRam} setRammn={setRammn} setRammx={setRammx} selects={selects} setSelects={setSelect}/>
+            </Flex>
+        </Flex>
+      </Container>
+    );
+  }
```

Comment:
It is not necessary that each configuration be a state. The global configuration object can be a single state. For example:

```javascript
useState({
  username: ....,
  memory: {
    max: ...,
    min: ...
  }
}
```